### PR TITLE
add file GreekPuctuationCheckRule.java in the el directory

### PR DESCRIPTION
**Pull request by VirginiaKostara**
I've added a new rule regarding the punctuation in Greek language. The class included extends the abstract class AbstractPunctuationCheckRule.java which is found in the languagetool-core directory.

### CODE REVIEW

**Comment by danielnaber** [comment]
Thanks, but I think this won't do anything unless it's linked from `Greek.getRelevantRules()`. Could you also add a unit test for it?

**Comment by VirginiaKostara** [comment]
Yes, sure. I will add a relevant unit test.

**Comment by VirginiaKostara** [comment]
@danielnaber If everything looks okay to you, would you merge the PR? Or, should I add/change something else?

File: languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  /*
+   * (non-Javadoc)
+   * 
+   * @see
+   * org.languagetool.rules.AbstractPunctuationCheckRule#isPunctsJoinOk
+   * (java.lang.String)
+   */
+  @Override
+  protected final boolean isPunctsJoinOk(String tokens) {
+	String punctCases = "\\,{1} |\\.{1} |:{1} |·{1} | -{1} * -{1} |\\.{3} |!{1} |;{1} "; //punctuation + [ ]
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java:48]: * Indentation
* I'm not sure I understand the `{1}`

File: languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * A rule that matches regular expressions regarding 
+ * punctuation in the greek language
+ * 
+ * @author 
+ */
+public class GreekPunctuationCheckRule extends AbstractPunctuationCheckRule {
+
+  public GreekPunctuationCheckRule(ResourceBundle messages) {
+    super(messages);
+
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java:35]: empty line

File: languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+   * (java.lang.String)
+   */
+  @Override
+  protected final boolean isPunctsJoinOk(String tokens) {
+	String punctCases = "\\,{1} |\\.{1} |:{1} |·{1} | -{1} * -{1} |\\.{3} |!{1} |;{1} "; //punctuation + [ ]
+	punctCases += "|\\,{1}\n|\\.{1}\n|:{1}\n|·{1}\n|\\.{3}\n|!{1}\n|;{1}\n"; //punctuation + [\n]
+    return tokens.matches(punctCases);
+  }
+
+  /*
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java:53]: I suggest removing comments like these, they take a lot of space and don't provide much value

File: languagetool-language-modules/el/src/test/GreekPunctuationCheckRuleTest.java [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+ */
+
+package org.languagetool.rules.el;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+
+public class GreekPunctuationCheckRuleTest {
+
+    GreekPunctuationCheckRule pcr = new GreekPunctuationCheckRule();
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/test/GreekPunctuationCheckRuleTest.java:27]: `private final`

File: languagetool-language-modules/el/src/test/GreekPunctuationCheckRuleTest.java [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+
+
+    @Test
+    public void test_givenNonPunctuationText1() {
+		String x = "this is text...";
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/test/GreekPunctuationCheckRuleTest.java:32]: * indentation
* tests like these don't need to be in their own method per test

File: languagetool-language-modules/el/src/test/GreekPunctuationCheckRuleTest.java [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+package org.languagetool.rules.el;
+import org.junit.Test;
+import static org.junit.Assert.*;
+
+
+public class GreekPunctuationCheckRuleTest {
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/test/GreekPunctuationCheckRuleTest.java:25]: There should also be a test that tests on the `match()` level, to make more clear what this rule matches and what it doesn't match

**Review by danielnaber — CHANGES_REQUESTED** [review]

File: languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java (hunk 1 above)
Comment (VirginiaKostara) [reply languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java]: {1} quantifies the preceding character to match 1 time.

**Comment by VirginiaKostara** [comment]
@danielnaber Thank you very much for your recommendations. I tried to include them all in my latest commit.

File: languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java (hunk 1 above)
Comment (danielnaber) [reply languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java]: Then why not leave it out, as `x` and `x{1}` then mean the same?

File: languagetool-language-modules/el/src/test/java/org/languagetool/rules/el/GreekPunctuationCheckRuleTest.java [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+import java.io.IOException;
+
+import static org.junit.Assert.assertEquals;
+
+public class GreekPunctuationCheckRuleTest {
+
+  @Test
+  public void testRule() throws IOException {
+		final GreekPunctuationCheckRule rule = new GreekPunctuationCheckRule(TestTools.getEnglishMessages());
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/test/java/org/languagetool/rules/el/GreekPunctuationCheckRuleTest.java:36]: the 2-space indentation seems to be missing here

File: languagetool-language-modules/el/src/test/java/org/languagetool/rules/el/GreekPunctuationCheckRuleTest.java [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+		
+		RuleMatch[] matches;
+		JLanguageTool lt = new JLanguageTool(new Greek());
+    
+		matches = rule.match(lt.getAnalyzedSentence("Καλημέρα, χρόνια πολλά!"));
+		assertEquals(0, matches.length);
+
+		matches = rule.match(lt.getAnalyzedSentence("Κάνω μία επεξήγηση - η εξήγηση είναι μέσα στις παύλες - και συνεχίζω. "));
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/test/java/org/languagetool/rules/el/GreekPunctuationCheckRuleTest.java:44]: shouldn't the whitespace at the end of the string be removed?

File: languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+   * (non-Javadoc)
+   * 
+   * @see
+   * org.languagetool.rules.AbstractPunctuationCheckRule#isPunctsJoinOk
+   * (java.lang.String)
+   */
+  @Override
+  protected final boolean isPunctsJoinOk(String tokens) {
+	  String punctCases = "\\,{1} |\\.{1} |:{1} |·{1} | -{1} * -{1} |\\.{3} |!{1} |;{1} ";
```
Comment (danielnaber) [inline languagetool-language-modules/el/src/main/java/org/languagetool/rules/el/GreekPunctuationCheckRule.java:43]: 2-space indentation missing

**Review by danielnaber — CHANGES_REQUESTED** [review]

**Comment by VirginiaKostara** [comment]
@danielnaber In the latest commits, a test case always fails. The important thing is that this shouldn't happen based on the Greek syntax. In fact, my implementation concerning this punctuation principle in the isPunctsJoinOk() method is ignored. That is the reason why I added an exact example pair but it also failed during the tests.

**Comment by danielnaber** [comment]
I'm sorry, but you'll need to debug this yourself, as I don't have enough time for that.

**Comment by danielnaber** [comment]
Closing old PRs that have a conflict

**Pull request closed without merging**